preprocessing/granular_intent.py

```diff
@@ -0,0 +1,101 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+# dictionary for different granularity of TOP data intent
+granular_intent = {
+        "high":{
+            "COMBINE": "combine",
+            "GET_CONTACT": "get_contact",
+            "GET_DIRECTIONS": "navigatin",
```

Comment:
Nit: should this be "navigation"?

Reply:
Corrected!

Reply:
Hm, I don't see your update -- do you have to grab some sort of snapshot?

Reply:
Updates are in the new commits at the bottom of the timeline!

Reply:
Ah I see it now, thank you!